src/main/java/org/traccar/sms/SnsSmsClient.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2021 Anton Tananaev (anton@traccar.org)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.traccar.sms;
+
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.services.sns.AmazonSNS;
+import com.amazonaws.services.sns.AmazonSNSClientBuilder;
+import com.amazonaws.services.sns.model.MessageAttributeValue;
+import com.amazonaws.services.sns.model.PublishRequest;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.traccar.Context;
+import org.traccar.config.Keys;
+import org.traccar.notification.MessageException;
+
+import java.util.HashMap;
+import java.util.Map;
+
+
+public class SnsSmsClient implements SmsManager {
+    private static final Logger LOGGER = LoggerFactory.getLogger(SnsSmsClient.class);
+
+    private final AmazonSNS snsClient;
+
+    public SnsSmsClient() {
+        if (Context.getConfig().getString(Keys.AWS_REGION) == null
+                || Context.getConfig().getString(Keys.AWS_ACCESS_KEY) == null
+                || Context.getConfig().getString(Keys.AWS_SECRET_KEY) == null) {
+            throw new RuntimeException("SNS Not Configured Properly. Please provide valid config.");
+        }
+        snsClient = awsSNSClient();
+    }
+
+    public AmazonSNS awsSNSClient() {
```

Comment:
What's the reason for this method?

Reply:
To abstract the AWS SNS client so that we can use just the same client in future if we need to add new feature and we can use the same method.

Reply:
This part doesn't really make sense to me. Please inline for now.